File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        leftPtr = 0
        rightPtr = len(nums) -1
        while leftPtr <= rightPtr:
            if nums[leftPtr] == val:
                nums[leftPtr] = nums[rightPtr]
                rightPtr -=1
            else:
                leftPtr +=1
        return leftPtr

                

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.go
func twoSum(numbers []int, target int) []int {
	leftPtr := 0
	rightPtr := len(numbers) - 1
	for leftPtr < rightPtr{
		currentSum := numbers[leftPtr] + numbers[rightPtr] 
		if currentSum < target {
			leftPtr++
		} else if currentSum > target {
			rightPtr--
		} else {
			return []int{leftPtr +1, rightPtr +1}
		}
	}
	return nil
}
